components/dashboard/src/settings/SelectAccountModal.tsx

```diff
@@ -0,0 +1,86 @@
+        props.close();
+    };
+
+    const continueWithOtherAccount = () => {
+        const accessControlUrl = gitpodHostUrl.asAccessControl().toString();
+
+        const loginUrl = gitpodHostUrl.withApi({
+            pathname: '/login',
+            search: `host=${props.otherUser.authHost}&returnTo=${encodeURIComponent(accessControlUrl)}`
+        }).toString();
+
+        const logoutUrl = gitpodHostUrl.withApi({
+            pathname: "/logout",
+            search: `returnTo=${encodeURIComponent(loginUrl)}`
+        }).toString();
+
+        window.location.href = logoutUrl;
+    };
+
+    return (<Modal visible={true} onClose={props.close}>
+    <h3 className="pb-2">Select Account</h3>
+    <div className="border-t border-b border-gray-200 mt-2 -mx-6 px-6 py-4">
+        <p className="pb-2 text-gray-500 text-base">You are attempting to authorize a provider that is already connected with your other account on Gitpod.</p>
+
+        <div className="mt-4 flex rounded-md w-full bg-gray-200 p-4 mx-auto">
+            <img className="w-4 h-4 m-1 ml-2 mr-4" src={info} />
+            <span>
+                Disconnect a provider in one of you accounts, if you like to continue with the other account.
+            </span>
+        </div>
+
+        <div className="mt-4 flex-grow flex flex-row justify-around align-center">
+            <SelectableCard className="w-2/5 h-56" title="Current Account" selected={useAccount === 'current'} onClick={() => setUseAccount('current')}>
+                <div className="flex-grow flex flex-col justify-center align-center">
+                    <img className="m-auto rounded-full w-24 h-24" src={props.currentUser.avatarUrl} alt={props.currentUser.name}/>
+                    <span className="m-auto text-gray-700 text-md font-semibold">{props.currentUser.authName}</span>
+                    <span className="m-auto text-gray-400 text-md">{props.currentUser.authHost}</span>
+                </div>
+            </SelectableCard>
+
+            <SelectableCard className="w-2/5 h-56" title="Other Account" selected={useAccount === 'other'} onClick={() => setUseAccount('other')}>
+                <div className="flex-grow flex flex-col justify-center align-center">
+                    <img className="m-auto rounded-full w-24 h-24" src={props.otherUser.avatarUrl} alt={props.otherUser.name}/>
+                    <span className="m-auto text-gray-700 text-md font-semibold">{props.otherUser.authName}</span>
+                    <span className="m-auto text-gray-400 text-md">{props.otherUser.authHost}</span>
+                </div>
+            </SelectableCard>
+        </div>
+
+
+    </div>
+
+    <div className="flex justify-end mt-6">
+        <button className={"ml-2 danger secondary"} onClick={() => {
+            if (useAccount === "other") {
+                continueWithOtherAccount();
+            } else {
+                continueWithCurrentAccount();
+            }
+        }}>Continue</button>
```

Comment:
1. in case we're seeing this on the /integrations page, the continue with current basically means stay here ✅ 
2. in case we're seeing this on creation of workspaces, then it's translated to visiting /integrations ✅ 

please verify

Reply:
Sounds like a good MVC. Let's do it.